prometheus.lua

```diff
@@ -829,25 +820,47 @@ function Prometheus:metric_data()
   local seen_metrics = {}
   local output = {}
   for _, key in ipairs(keys) do
+    utils.yield(true)
+
     local value, err = self.dict:get(key)
     if value then
-      local short_name = short_metric_name(key)
-      if not seen_metrics[short_name] then
+      local match, err = ngx_re_match(key, METRICS_KEY_REGEX, "jo")
+
+      if err then
+        ngx.log(ngx.ERR, "failed to match regex: ", err)
+        return
+      end
```

Comment:
Is there any reason why you've moved this out of `fix_histogram_bucket_labels` into the loop here? It still seems to be called for every loop iteration.

Reply:
restored